### PR TITLE
Compat with latest VLLM 0.4.2 main + fork.number rename + Flashinfer 0.0.4

### PR DESCRIPTION
**Pull request by Qubitium**
Reason for PR:

* Compat with VLLM main
* Rename `fork(number=N)` param to `fork(size=N)` for clarity. 
* Complete flashinfer 0.0.4 todo

### CODE REVIEW

File: python/sglang/srt/weight_utils.py [hunk 1]
```diff
@@ -0,0 +1,397 @@
+"""Utilities for downloading and initializing model weights."""
+import fnmatch
```
Comment (merrymercy) [inline python/sglang/srt/weight_utils.py:7]: Where is this file from? Are there any references? If so, could you put some links to the source of code here?

File: python/sglang/srt/managers/router/model_runner.py [hunk 2]
```diff
@@ -40,6 +40,10 @@ def import_model_classes():
     package = importlib.import_module(package_name)
     for _, name, ispkg in pkgutil.iter_modules(package.__path__, package_name + "."):
         if not ispkg:
+            # gemma references InputMetadata that does not exist, skip it for now.
+            # TODO Wait for sglang/main to fix it.
+            if name in ["sglang.srt.models.gemma"]:
+                continue
```
Comment (merrymercy) [inline python/sglang/srt/managers/router/model_runner.py]: What are the problems? Could you tell me how to fix it?

**Review by merrymercy — CHANGES_REQUESTED** [review]

**Comment by merrymercy** [comment]
@Qubitium Thanks for the contributions! We would like to merge this soon.

**Comment by Qubitium** [comment]
@merrymercy  The vllm main is moving quite fast. We have not tested main compatibility as of today and staff is on a 5 day vacation so response will be delayed.

File: python/sglang/srt/weight_utils.py (hunk 1 above)
Comment (Qubitium) [reply python/sglang/srt/weight_utils.py:7]: @ZhouXingg  Please add comment that this is code is copied/adapted from vllm before their refractor causing sglang compat issues.

File: python/sglang/srt/managers/router/model_runner.py (hunk 2 above)
Comment (Qubitium) [reply python/sglang/srt/managers/router/model_runner.py]: @ZhouXingg

**Comment by Qubitium** [comment]
@merrymercy  Remerged with vllm main as of today and confirmed working.

**Review by merrymercy — APPROVED** [review]

**Comment by merrymercy** [comment]
@Qubitium @ZhouXingg Thanks!

**Pull request merged**